geniusyield-market-maker/src/GeniusYield/MarketMaker/MakerBot.hs

```diff
@@ -115,19 +130,129 @@ buildAndSubmitActions user@User {uColl, uStakeCred} providers netId ua di = flip
+
+mbStateMachine ∷ Strategy
+               → MakerBot
+               → GYNetworkId
+               → GYProviders
+               → PricesProviders
+               → DEXInfo
+               → MBFret
+               → StateT MBFret IO ()
+mbStateMachine runStrategy mb@MakerBot {mbUser, mbDelay, mbToken} netId providers pp di mbFret = do
+  let cfg          = ppaCommonCfg . pricesAggregatorPP $ pp
+      respiteDelay = ceiling $ pccRespiteDelayFactor cfg * fromIntegral mbDelay
+
+  case mbFret of
+    MBReady → do
+      (newActions, controller) ← lift $ runStrategy pp mbUser mbToken
+      case controller of
+        UACNormal     → do
+          lift $ buildAndSubmitActions mbUser providers netId newActions di
+
+          lift $ gyLogInfo providers logNS "---------- Done for the block! -----------"
+          lift $ threadDelay mbDelay
+
+        UACSpooked1   → do
+          put MBSpooked1 { mbs1Relax = 0, mbs1Worse = 0 }
+          lift $ gyLogInfo providers logNS "Closing all orders due to price mismatch among Prices Providers"
+
+          lift $ cancelAllOrders' mb netId providers di
+          lift $ threadDelay respiteDelay
+
+        UACSpooked2 e → do
+          put MBSpooked2 { mbs2Relax = 0 }
+          lift $ gyLogWarning providers logNS $ "Closing all orders due to: " ++ e
+
+          lift $ cancelAllOrders' mb netId providers di
+          lift $ threadDelay respiteDelay
+
+    MBSpooked1 {..} → do
+      if mbs1Relax > pccAfterExitRelaxAim1 cfg then do
+           lift $ gyLogInfo providers logNS $ "Resuming strategy"
+           put MBReady
+         else if mbs1Worse > pccAfterExitWorseMax1 cfg
+                 then do
+                   lift $ gyLogWarning providers logNS $ "Price mismatch among providers lasting a long time!"
+                   put MBSpooked2 { mbs2Relax = 0 }
+                 else do
+                   pe ← lift $ priceEstimate' pp mbToken
+
+                   case pe of
+                     PriceMismatch1    → do
+                       lift $ gyLogInfo providers logNS $ "Price mismatch persists"
+                       put MBSpooked1 { mbs1Relax = 0, mbs1Worse = mbs1Worse + 1 }
+
+                     PriceMismatch2    → do
+                       lift $ gyLogWarning providers logNS $ "Outrageous price mismatch among providers!"
+                       put MBSpooked2 { mbs2Relax = 0 }
+
+                     PriceUnavailable  → do
+                       lift $ gyLogWarning providers logNS $ "All Prices Providers unavailable!"
+                       put MBSpooked2 { mbs2Relax = 0 }
+
+                     psf@PriceSourceFail {} → do
+                       lift $ logPricesProviderFail providers psf
+
+                     _                 → do
+                       lift $ gyLogInfo providers logNS $ "Apparent recovery of Prices Providers; waiting for observation period to elapse."
+                       put MBSpooked1 { mbs1Relax = mbs1Relax + 1, mbs1Worse = mbs1Worse }
```

Comment:
Can omit `mbs1Worse = mbs1Worse`

Reply:
Done.

Reply:
Might be missing an extension, but fine.